extensions/azurecore/src/account-provider/utils/simpleWebServer.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as http from 'http';
+import * as url from 'url';
+import { AddressInfo } from 'net';
+
+export type WebHandler = (req: http.IncomingMessage, reqUrl: url.UrlWithParsedQuery, res: http.ServerResponse) => void;
+
+export class SimpleWebServer {
+	private readonly pathMappings = new Map<string, WebHandler>();
+	private readonly server: http.Server;
+	private lastUsed: number;
+	private shutoffInterval: NodeJS.Timer;
+
+	constructor(private readonly autoShutoffTimer = 5 * 60 * 1000) { // Default to five minutes.
+		this.bumpLastUsed();
+		this.autoShutoff();
+		this.server = http.createServer((req, res) => {
+			this.bumpLastUsed();
+			const reqUrl = url.parse(req.url!, /* parseQueryString */ true);
+
+			const handler = this.pathMappings.get(reqUrl.pathname);
+			if (handler) {
+				return handler(req, reqUrl, res);
+			}
+
+			console.error('Unhandled request ', reqUrl);
+		});
+	}
+
+	private bumpLastUsed(): void {
+		this.lastUsed = new Date().getTime();
+	}
+
+	public async shutdown(): Promise<void> {
+		clearInterval(this.shutoffInterval);
+		return new Promise<void>((resolve, reject) => {
+			this.server.close((error) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve();
+				}
+			});
+		});
+	}
+
+	public startup(): Promise<string> {
+		let portTimeout: NodeJS.Timer;
```

Comment:
What's the behavior if startup is called more than once? May want to check that here before setting all this up.

Reply:
Good catch, we shouldn't do that but i'll put a check